src/linter/engine.mjs

```diff
@@ -10,13 +10,15 @@ const createLinterEngine = rules => {
    * Validates a ApiDocMetadataEntry entry against all defined rules
    *
    * @param {ApiDocMetadataEntry} entry
+   * @param {import('./types').LintDeclarations}
+   * @param declarations
    * @returns {import('./types').LintIssue[]}
    */
-  const lint = entry => {
+  const lint = (entry, declarations) => {
```

Comment:
The declarations kinda feel retrofitted in but not sure if there's a better way to pass them

Reply:
I think declarations could be part of metadata entries